src/core/src/core_logic/PatchAssessor.py

```diff
@@ -81,15 +88,23 @@ def start_assessment(self):
                 else:
                     error_msg = 'Error retrieving available patches: ' + repr(error)
                     self.composite_logger.log_error(error_msg)
+                    self.write_assessment_perf_logs(number_of_tries, Constants.LogStrings.FAILED, error_msg)
                     self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
                     if Constants.ERROR_ADDED_TO_STATUS not in repr(error):
                         error.args = (error.args, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
                     self.status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
                     raise
 
+        self.write_assessment_perf_logs(number_of_tries, Constants.LogStrings.SUCCEEDED, "")
         self.composite_logger.log("\nPatch assessment completed.\n")
         return True
 
+    def write_assessment_perf_logs(self, number_of_tries, task_status, error_msg):
+        assessment_perf_log = {Constants.LogStrings.TASK: Constants.ASSESSMENT, Constants.LogStrings.PACKAGE_MANAGER: self.package_manager_name,
+                               Constants.LogStrings.NUMBER_OF_TRIALS: str(number_of_tries), Constants.LogStrings.TASK_STATUS: task_status,
```

Comment:
same comments as in Patchinstaller about keeping Task and Task_Status together

Reply:
Agree. I think error_msg should be together with task_status so should I move error_msg also with task_status?

Reply:
Agreed and I see you have done this already in both PatchAssessor and PatchInstaller. Resolving this comment